importer/Uploader.py

```diff
@@ -21,13 +21,56 @@ def add_labels(self, target_item, labels, log):
             message = "{} ADDED LABELS {}".format(t_id, labels)
             log.logit(message)
 
-    def add_descriptions(self, target_item, descriptions, log):
+    def add_descriptions(self, target_item, descriptions, log,
+                         disambiguated=None):
         """Add descriptions."""
-        self.wdstuff.add_multiple_descriptions(descriptions, target_item)
-        if log:
-            t_id = target_item.getID()
-            message = "{} ADDED DESCRIPTIONS {}".format(t_id, descriptions)
-            log.logit(message)
+        disambiguated = disambiguated or []
+        try:
+            self.wdstuff.add_multiple_descriptions(descriptions, target_item)
+        except pywikibot.exceptions.OtherPageSaveError as e:
+            new_desc = self.add_disambiguators_to_descriptions(
+                descriptions, e, disambiguated)
+            if new_desc:
+                self.add_descriptions(
+                    target_item, new_desc, log, disambiguated)
+            else:
+                raise
+        else:
+            if log:
+                t_id = target_item.getID()
+                message = "{} ADDED DESCRIPTIONS {}".format(t_id, descriptions)
+                log.logit(message)
+
+    def add_disambiguators_to_descriptions(self, descriptions, api_error,
+                                           disambiguated):
+        """
+        Add needed disambiguators to descriptions.
+
+        :param descriptions: pre-existing descriptions
+        :param api_error: the raised pywikibot.data.api.APIError
```

Comment:
Can be updated now that it works since we're not passing an APIError directly here.

Reply:
Done